### ISSUE TITLE
Limit results

### ISSUE THREAD

**Issue by LangLangBart** [gennaro-tedesco/gh-s#17]
## Description
A QoL addition would be an extra flag to limit the number of results, as the options at the top are not always visible due to my small window size. I tried something, see the diff and GIF below for more clarity. I can open a PR, but open an issue first to get feedback on whether this is something you actually want to include in your project, and to ask if you even want to write the code yourself (I'm not a `Go` programmer, but it seems doable).

### Diff
```diff
--- a/cmd/root.go
+++ b/cmd/root.go
@@ -28,6 +28,7 @@ var rootCmd = &cobra.Command{
 		user, _ := cmd.Flags().GetString("user")
 		topicList, _ := cmd.Flags().GetStringSlice("topic")
 		colour, _ := cmd.Flags().GetString("colour")
+		limit, _ := cmd.Flags().GetInt("limit")

 		searchString := func() string {
 			if empty, _ := (cmd.Flags().GetBool("empty")); empty {
@@ -45,7 +46,7 @@ var rootCmd = &cobra.Command{
 			fmt.Println("\033[31m ✘\033[0m No results found")
 			os.Exit(1)
 		}
-		PromptList := getSelectionPrompt(repos, colour)
+		PromptList := getSelectionPrompt(repos, colour, limit)

 		idx, _, err := PromptList.Run()
 		if err != nil {
@@ -68,6 +69,7 @@ func init() {
 	rootCmd.Flags().StringP("user", "u", "", "search repository by user")
 	rootCmd.Flags().StringSliceVarP(&topics, "topic", "t", []string{}, "search repository by topic")
 	rootCmd.Flags().StringP("colour", "c", "cyan", "colour of selection prompt")
+	rootCmd.Flags().IntP("limit", "L", 20, "Limit results")
 	rootCmd.Flags().BoolP("empty", "E", false, "allow for empty name search")
 	rootCmd.Flags().BoolP("version", "V", false, "print current version")
 	rootCmd.SetHelpTemplate(getRootHelp())
@@ -126,6 +128,7 @@ Flags:
                 multiple topics can be specified:
                 -t go -t gh-extension
   -c, --colour  change prompt colour
+  -L, --limit   Limit results
   -V, --version print current version
   -h, --help    show this help page

--- a/cmd/ui.go
+++ b/cmd/ui.go
@@ -85,14 +85,14 @@ func getTemplate(colour string) *promptui.SelectTemplates {

 }

-func getSelectionPrompt(repos []repoInfo, colour string) *promptui.Select {
+func getSelectionPrompt(repos []repoInfo, colour string, limit int) *promptui.Select {
 	return &promptui.Select{
 		Stdout:    os.Stderr,
 		Stdin:     os.Stdin,
 		Label:     "repository list",
 		Items:     repos,
 		Templates: getTemplate(colour),
-		Size:      20,
+		Size:      limit,
 		Searcher: func(input string, idx int) bool {
 			repo := repos[idx]
 			title := strings.ToLower(repo.Name)

```
### GIF
First the standard command `gh s awesome` and then with the additional `-L` flag.
<img src="https://ttm.sh/w97.gif" width="600">

**Comment by gennaro-tedesco** [comment]
Thank you for the idea, I would say this is definitely something that can be included, I do not see a counterargument: whilst I presume the majority of people have normal terminal sizes and will not benefit from it, at the same time it does not hurt the scope.

If you already have an implementation (which I see from your code above) feel free to open a PR!

**Comment by gennaro-tedesco** [comment]
Thank you for your contribution! The limit flag is available in the [0.0.7](https://github.com/gennaro-tedesco/gh-s/releases/tag/v0.0.7) release!

**Comment by LangLangBart** [comment]
Thank you for the commit, good tools you have created for the Github cli community.
